sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs

```diff
@@ -12,7 +12,13 @@ public DiagnosticsOptions()
         {
             IsTelemetryEnabled = !EnvironmentVariableToBool(Environment.GetEnvironmentVariable("AZURE_TELEMETRY_DISABLED")) ?? true;
             ApplicationId = DefaultApplicationId;
-            LoggingAllowedHeaderNames = new List<string>();
+            LoggingAllowedHeaderNames = new List<string>()
```

Comment:
Nit: Sort these headers.

Reply:
Or should it just be a HashSet if we think enough vales to warrant the overhead for O(1) lookup is worth it?

Reply:
We can't use HashSet here because we are not materializing strings in redacting code. If overhead becomes too high we can invent something custom to make things faster, for now, I'd like to see if anyone notices.